modules/cnf-provisioning-real-time-and-low-latency-workloads.adoc

```diff
@@ -325,4 +325,115 @@ spec:
 [source,terminal]
 ----
 $ oc apply -f my-performance-profile.yaml
-----
\ No newline at end of file
+----
+
+[id="node-tuning-operator-pod-power-saving-config_{context}"]
+== Optional: Power saving configurations for pods
+
+You can enable power savings for low priority workloads that are co-located with high priority workloads on an SNO node without impacting the latency or throughput of the high priority workloads. Power saving is possible without modifications to the workloads themselves.
+
+The power saving configurations apply at the pod level, which means that the configuration applies to all the cores used by the pod.
+
+By enabling P-states and C-states, high priority workloads can be configured for best performance and lowest latency in low power mode.
+
+.Power saving configurations
+[cols="1,2", options="header"]
+|====
+|Annotation
+|Description
+
+|`low-latency.crio.io: "enable"`
+|Enables low latency and high performance.
+
+a|[source,terminal]
+----
+annotations:
+`cpu-c-states.crio.io: "enable"`
+`cpu-freq-governor.crio.io: "<governor>"`
+----
+|Allows for flexibility by enabling the lowest C-state in the pod and specifying the governor type for CPU scaling.
+|====
+
+
+.Prerequisites
+
+* You enabled C-states and P-states in the BIOS
+
+.Procedure
+
+. Generate a `PerformanceProfile` with `per-pod-power-management` set to `true`:
++
+[source,terminal]
+----
+$ podman run --entrypoint performance-profile-creator -v \
+/must-gather:/must-gather:z registry.redhat.io/openshift4/performance-addon-rhel8-operator:v4.11 \
+--mcp-name=worker-cnf --reserved-cpu-count=20 --rt-kernel=true \
+--split-reserved-cpus-across-numa=false --topology-manager-policy=single-numa-node \
+--must-gather-dir-path /must-gather -power-consumption-mode=low-latency \ <1>
+--per-pod-power-management=true > my-performance-profile.yaml
+----
+<1> The `power-consumption-mode` in combination with enabled `per-pod-power-management` must be set to `default` or `low-latency`.
+
++
+.Example PerformanceProfile with perPodPowerManagement
+
+[source,yaml]
+----
+apiVersion: performance.openshift.io/v2
+kind: PerformanceProfile
+metadata:
+     name: performance
+spec:
+    [.....]
+    workloadHints:
+        realTime: true
+        highPowerConsumption: false
+        perPodPowerManagement: true
+----
+
+. Set the default `cpufreq` governor with the kernel cmdline option in the `PerformanceProfile` CR:
++
+[source,yaml]
+----
+apiVersion: performance.openshift.io/v2
+kind: PerformanceProfile
+metadata:
+     name: performance
+spec:
+    [.....]
+    additionalKernelArgs:
+    - cpufreq.default_governor=schedutil
```

Comment:
Probably want a node for this to say "schedutil" is recommended, but other governors can be used (e.g. ondemand, powersave).